spec/integration/issues_spec.rb

```diff
@@ -0,0 +1,100 @@
+# frozen_string_literal: true
+
+RSpec.describe "GitHub issues" do
+  describe "dry-rb/dry-validation#718" do
```

Comment:
I have git blame for that. I'm not against having some real-world use cases but we need more targeted specs as well. The problem with specs like here is if you work on something and it breaks it can be really hard to find the cause